### ISSUE TITLE
[Angular] Upgrading to 10: Object(_nativescript_core__WEBPACK_IMPORTED_MODULE_0__["profile"]) is not a function

### ISSUE THREAD

**Issue by Codex-** [NativeScript/NativeScript#8953]
**Environment**
Provide version numbers for the following components (information can be retrieved by running `tns info` in your project folder or by inspecting the `package.json` of the project):

-   CLI: 7.0.10
-   Cross-platform modules: 6.5.20
-   Android Runtime: 6.5.3
-   iOS Runtime: 6.5.3
-   Plugin(s):

Deps:

```
"@angular/animations": "~10.0.0",
"@angular/common": "~10.0.0",
"@angular/compiler": "~10.0.0",
"@angular/core": "~10.0.0",
"@angular/forms": "~10.0.0",
"@angular/platform-browser-dynamic": "~10.0.0",
"@angular/platform-browser": "~10.0.0",
"@angular/router": "~10.0.0",
"@nativescript/angular": "~10.0.0",
"@nativescript/core": "^6.5.19",
"@ngx-translate/core": "^13.0.0",
"@nstudio/nativescript-camera-plus": "^3.0.7",
"@nstudio/nativescript-cardview": "^1.0.0",
"@nstudio/nativescript-checkbox": "^2.0.2",
"@nstudio/nativescript-loading-indicator": "^3.0.3",
"@nstudio/nativescript-mixpanel": "^1.0.0",
"@nstudio/nativescript-pulltorefresh": "^1.1.1",
"nativescript-barcodescanner": "^3.4.2",
"nativescript-camera": "^4.5.0",
"nativescript-drawingpad": "^3.1.0",
"nativescript-geolocation": "^5.1.0",
"nativescript-iqkeyboardmanager": "^1.5.1",
"nativescript-permissions": "^1.3.8",
"nativescript-plugin-firebase": "^10.5.0",
"nativescript-ripple": "^4.0.1",
"nativescript-secure-storage": "^2.6.0",
"tns-core-modules": "~6.5.20",
```

Dev Deps:

```
"@angular-devkit/build-angular": "~0.1000.2",
"@angular/cli": "~10.0.0",
"@angular/compiler-cli": "~10.0.0",
"@nativescript/webpack": "~2.1.3",
"@ngtools/webpack": "~10.0.0",
"tns-android": "6.5.3",
"tns-ios": "^6.5.3",
"tns-platform-declarations": "~6.5.15",
"typescript": "~3.9.0",
```

package:

```json
{
	"name": "x",
	"main": "main.js",
	"version": "1.0.0",
	"android": {
		"v8Flags": "--expose_gc",
		"markingMode": "none"
	},
	"cssParser": "rework"
}
```

**Describe the bug**

I followed the general Angular 10 upgrade instructions in addition to the upgrade guides provided in the blog. However at runtime at launch this error occurs:

```
ns run ios

...

file: node_modules/@nativescript/angular/fesm2015/nativescript-angular.js:2946:35: JS ERROR TypeError: Object(_nativescript_core__WEBPACK_IMPORTED_MODULE_0__["profile"]) is not a function. (In 'Object(_nativescript_core__WEBPACK_IMPORTED_MODULE_0__["profile"])(`"renderer".addScopedStyleToCss`, ɵ1$2)', 'Object(_nativescript_core__WEBPACK_IMPORTED_MODULE_0__["profile"])' is an instance of Object)
(CoreFoundation) *** Terminating app due to uncaught exception 'NativeScript encountered a fatal error: TypeError: Object(_nativescript_core__WEBPACK_IMPORTED_MODULE_0__["profile"]) is not a function. (In 'Object(_nativescript_core__WEBPACK_IMPORTED_MODULE_0__["profile"])(`"renderer".addScopedStyleToCss`, …µ1$2)', 'Object(_nativescript_core__WEBPACK_IMPORTED_MODULE_0__["profile"])' is an instance of Object)
```

```
ns run android

...

System.err: An uncaught Exception occurred on "main" thread.
System.err: Unable to create application com.tns.NativeScriptApplication: com.tns.NativeScriptException: Error calling module function
System.err: TypeError: Object(...) is not a function
System.err: File: (file: node_modules/@nativescript/angular/fesm2015/nativescript-angular.js:2946:35)
System.err:
System.err: StackTrace:
System.err: (file: node_modules/@nativescript/angular/fesm2015/nativescript-angular.js:2946:35)
System.err:     at ../node_modules/@nativescript/angular/fesm2015/nativescript-angular.js(file:///data/data/org.nativescript.n65sample/files/app/vendor.js:95754:30)
System.err:     at __webpack_require__(file: src/webpack/bootstrap:816:0)
System.err:     at fn(file: src/webpack/bootstrap:120:0)
System.err:     at (file:///data/data/org.nativescript.n65sample/files/app/bundle.js:329:125)
System.err:     at ./main.ts(file:///data/data/org.nativescript.n65sample/files/app/bundle.js:396:30)
System.err:     at __webpack_require__(file: src/webpack/bootstrap:816:0)
System.err:     at checkDeferredModules(file: src/webpack/bootstrap:43:0)
System.err:     at webpackJsonpCallback(file: src/webpack/bootstrap:30:0)
System.err:     at (file:///data/data/org.nativescript.n65sample/files/app/bundle.js:2:57)
System.err:     at require(:1:266)
System.err:
System.err:
System.err: TypeError: Object(...) is not a function
System.err:
System.err: StackTrace:
System.err: java.lang.RuntimeException: Unable to create application com.tns.NativeScriptApplication: com.tns.NativeScriptException: Error calling module function
System.err: TypeError: Object(...) is not a function
System.err: File: (file: node_modules/@nativescript/angular/fesm2015/nativescript-angular.js:2946:35)
System.err:
System.err: StackTrace:
System.err: (file: node_modules/@nativescript/angular/fesm2015/nativescript-angular.js:2946:35)
System.err:     at ../node_modules/@nativescript/angular/fesm2015/nativescript-angular.js(file:///data/data/org.nativescript.n65sample/files/app/vendor.js:95754:30)
System.err:     at __webpack_require__(file: src/webpack/bootstrap:816:0)
System.err:     at fn(file: src/webpack/bootstrap:120:0)
System.err:     at (file:///data/data/org.nativescript.n65sample/files/app/bundle.js:329:125)
System.err:     at ./main.ts(file:///data/data/org.nativescript.n65sample/files/app/bundle.js:396:30)
System.err:     at __webpack_require__(file: src/webpack/bootstrap:816:0)
System.err:     at checkDeferredModules(file: src/webpack/bootstrap:43:0)
System.err:     at webpackJsonpCallback(file: src/webpack/bootstrap:30:0)
System.err:     at (file:///data/data/org.nativescript.n65sample/files/app/bundle.js:2:57)
System.err:     at require(:1:266)
System.err:
System.err:
System.err: TypeError: Object(...) is not a function
System.err:     at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6465)
System.err:     at android.app.ActivityThread.access$1300(ActivityThread.java:219)
System.err:     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859)
System.err:     at android.os.Handler.dispatchMessage(Handler.java:107)
System.err:     at android.os.Looper.loop(Looper.java:214)
System.err:     at android.app.ActivityThread.main(ActivityThread.java:7356)
System.err:     at java.lang.reflect.Method.invoke(Native Method)
System.err:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
System.err: Caused by: com.tns.NativeScriptException: Error calling module function
System.err: TypeError: Object(...) is not a function
System.err: File: (file: node_modules/@nativescript/angular/fesm2015/nativescript-angular.js:2946:35)
System.err:
System.err: StackTrace:
System.err: (file: node_modules/@nativescript/angular/fesm2015/nativescript-angular.js:2946:35)
System.err:     at ../node_modules/@nativescript/angular/fesm2015/nativescript-angular.js(file:///data/data/org.nativescript.n65sample/files/app/vendor.js:95754:30)
System.err:     at __webpack_require__(file: src/webpack/bootstrap:816:0)
System.err:     at fn(file: src/webpack/bootstrap:120:0)
System.err:     at (file:///data/data/org.nativescript.n65sample/files/app/bundle.js:329:125)
System.err:     at ./main.ts(file:///data/data/org.nativescript.n65sample/files/app/bundle.js:396:30)
System.err:     at __webpack_require__(file: src/webpack/bootstrap:816:0)
System.err:     at checkDeferredModules(file: src/webpack/bootstrap:43:0)
System.err:     at webpackJsonpCallback(file: src/webpack/bootstrap:30:0)
System.err:     at (file:///data/data/org.nativescript.n65sample/files/app/bundle.js:2:57)
System.err:     at require(:1:266)
System.err:
System.err:
System.err: TypeError: Object(...) is not a function
System.err:     at com.tns.Runtime.runModule(Native Method)
System.err:     at com.tns.Runtime.runModule(Runtime.java:674)
System.err:     at com.tns.Runtime.run(Runtime.java:666)
System.err:     at com.tns.NativeScriptApplication.onCreate(NativeScriptApplication.java:21)
System.err:     at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1182)
System.err:     at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6460)
System.err:     ... 8 more
```

I've attempted many things such as starting the upgrade steps again from scratch, locking packages, clearing locks, all sorts of things and keep coming up empty.

**To Reproduce**

Upgrade an Angular 9 NS6.5 project to Angular 10.

**Expected behavior**
Launch without crashing :(

**Sample project**

I've created a sample repo that shows the issue here: https://github.com/Codex-/ns65-angular10-crash

**Comment by NathanWalker** [comment]
@Codex- Here's a PR with the proper updates:
https://github.com/Codex-/ns65-angular10-crash/pull/1

There were several dependencies's which were incorrect and not reflective of the update guide here: https://nativescript.org/blog/nativescript-7-announcement/

Hope the PR referenced helps others in similar situation whom come across the closed ticket.

**Comment by Codex-** [comment]
Sorry I feel like I was not clear enough, we're not wanting to move to N7 yet. There are several concerns with the current version that ultimately have lead to delaying moving to N7 for now (however ios14 forced my hand with the launch crashes) and among many things the new config structure is not compatible without large changes to CICD for me.

I'm also in the middle of upgrading the nativescript-mixpanel plugin to support n7 :P

The goal was to migrate to Angular 10 on the latest 6.* stable, and migrating in a separate work effort.

**Comment by NathanaelA** [comment]
@Codex- Angular 9&10 do not support NS 6 (and probably never will).   If you need to use NS6, you have to stay with the much older Angular.   However, in NS6, you can use the latest NS6 IOS 6.5.2 runtime and it works for IOS 14.

**Comment by Codex-** [comment]
> @Codex- Angular 9&10 do not support NS 6 (and probably never will). If you need to use NS6, you have to stay with the much older Angular. However, for NS6, you can use the latest NS6 IOS 6.5.2 runtime and it works for IOS 14, so staying with NS6 and having iOS 14 support is simple.

I must have misinterpreted the [blog post](https://nativescript.org/blog/upgrading-tips-for-angular-10/) since the [example](https://github.com/mahmoudajawad/nativescript-pokedex/commit/2e9723dd093244761cd524cb068a0563d5813e22) is an NS6 project and the posts talk about preparing your project for NS7 (and also predates the 7 release).

fwiw `tns-ios@6.5.3` seems to be stable too and actually launches.

> However, it is also possible that you also need to stay with XCode 11 with the NS6 tooling. We found several things we had to change in the tooling to make XCode 12 work properly with NS7 which weren't exactly backwards compatible..

Yeah ns6 with xcode 11 wasn't an option, crash on launch on various iphone devices (11, 11 pro, xr). However, works with latest 6.5.* and xcode 12_2_b2.

**Comment by NathanaelA** [comment]
Yeah, sorry meant 6.5.3 (also fixed the post above)

We still support 6.5.x; which is why we released the iOS 6.5.3 engine to allow everyone to easily be able to deploy to IOS 14 with their app without upgrading.    Just all the changes to upgrade Angular 9/10 to actually work in NS are part of what made NS7, NS7.   

To my knowledge the full 6.5.x tool chain is still solid as long as you are using XCode 11 and the NS 6 CLI.   If you are using the NS7 cli, then you pretty much have to use XCode 12.  

We are also unaware of any crashes on IPhone 11/pro/xr using NS 6 cli as long as you are using the 6.5.3 tns-ios engine.   If you are using a earlier tns-ios, you will most certainly see issue on any ios 14 device.

Just a FYI, Using XCode 12.2 your apps will be rejected by Apple, you can't release a app to the store using beta software.   However if you are attempting to use NS7 CLI you pretty much should use XCode 12.01.
